docs/vagrant.md

```diff
@@ -0,0 +1,102 @@
+
+## Prerequisites
+
+1. You will require 4 GB of memory.
+1. You will require 8 GB of disk space.
+1. `npm install` creates symbolic links. As a result, the VM user: `vagrant` requires `create symbolic link` authority. In some environements, only the administrator has this authority. To resolve this issue in Windows, the following steps can be performed:
+   1. Launch the `Local Security Policy` app from the `start menu` (or `Win+R`, then type `secpol.msc`)
+   1. Navigate to `Local Policies` -> `User Rights Assignment`.
+   1. Open the `Create symbolic links` property. By default it has only `Administrators` listed.
+   1. Click `Add User or Group...` button and type `Authenticated Users` in the `Enter the object names to select` field and then press the `OK` button. This will grant `symbolic link` priviledge to all users who log on with credetials.
+   1. Close any open windows by clicking `OKs` and and then `log off` and `log on` to activate.
+   1. Symbolic links should now work in VirtualBox shared folders without administrator priviledges.
+
+## Setup Steps
+
+1. [Download](https://developer.hashicorp.com/vagrant/downloads?ajs_aid=ba208f95-d5a1-457d-abeb-49d458b95eec&product_intent=vagrant) Vagrant software.
+1. Reference these [installation](https://developer.hashicorp.com/vagrant/docs/installation) instructions.
+1. Verify installation by running `vagrant -v` in a terminal session.
+1. Install [VirtualBox](https://www.virtualbox.org/).
+1. If you have not already done so, fork repo to your local machine.
+1. Next you will need to install a [box](https://developer.hashicorp.com/vagrant/tutorials/getting-started/getting-started-boxes). The `Vagrantfile` in this repository uses box: `generic/ubuntu2204`. So to install this box, run in a terminal window:
+
+```bash
+    vagrant box add generic/ubuntu2204
+```
+
+To see which boxes are installed run:
+
+```bash
+    vagrant box list
+```
+
+7. Create a new folder and name it something that identifies the VM you are about to create. Example: `UbuntuVM`. The name of this folder will be used by Vagrant to identify this specific VM. It is to this directory that you will copy the vagrant file we have created for you. In a later step, Vagrant will create subdirectory`.vagrant` which it will use to manage your VM. It is in this directory where you will run all Vagrant commands that are specific to your VM.
+8. Copy file: `Vagrantfile` from host `gloss-translation` repository `.vagrant/` folder to the new folder you created above.
+9. Read the `NOTES:` section contained in file `Vagrantfile`.
+10. in the `Vagrantfile`, edit `HOST_FOLDER` by replacing `../gloss-translation` with the location of your repository root folder.
+
+```ruby
+    HOST_FOLDER = "../gloss-translation"
+```
+
+11. If required, perform any requested `plugin` installations mentioned in the `NOTES:` section. To see which plugins are installed run:
```

Comment:
Your recommendation to put Vagrantfile in the root directory is fine with me and I see the benefit of it.  However, I also see a downside in that then, we could only support a single provider. This is because we could only support a single version of the Vagrantfile.  

I have already created and successfully tested another version of the Vagrantfile that creates a VM using docker.  If you want to offer contributors the option to choose between multiple providers we would have to come up with a different approach. This was the reason why my design had the contributor copy the Vagrantfile of their choice to the VM folder.

I am good with whatever you wish to do.  Please let me know if you want to stick with offering only VirtualBox, or reconfigure with the  option to choose between multiple providers.

Reply:
[This from the vagrant ](https://developer.hashicorp.com/vagrant/docs/providers/configuration)docs suggests that multiple providers is not an issue. These lines from the vagrantfile are ignored if you use a different provider:
```ruby
config.vm.provider "virtualbox" do |vb|
    vb.memory = 4096  # Set RAM in MB (4GB in this example)
    vb.cpus = 2      # Set the number of CPU cores
    vb.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate//home/vagrant/vmrepo", "1"]
  end
```

Reply:
Thanks for pointing that out to me.  Was not aware of this.  So I will update Vagrantfile so it is compatible with both VirtualBox and Docker. Then will implement all of your recommendations.  When complete will request another review.